### PR TITLE
Feature/enable tag suffix

### PR DESCRIPTION
**Pull request by ghostsquad**
The usage of the params of this resource felt a little off, so I thought I'd add some functionality, clean up the scripts a bit and rename params to be more clear (while maintaining backwards compatibility).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,16 +174,24 @@ version is the image's digest.
...
+
+* `tag`: **DEPRECATED. Use `tag_file` instead**
+* `tag_file`: *Optional.* The value should be a path to a file containing the name
+  of the tag. Will be prefixed/suffixed if those params are present.
 
 * `tag_prefix`: *Optional.* If specified, the tag read from the file will be
   prepended with this string. This is useful for adding `v` in front of version
   numbers.
 
+* `tag_suffix`: *Optional.* Adds a suffix to the tag. Useful for things like `-$BUILD_ID`
```
Comment (vito) [inline README.md]: Does `-$BUILD_ID` actually work? I don't see where it'd be interpolated.

Comment (ghostsquad) [reply README.md]: this is talked about here:

**Comment by ghostsquad** [comment]
@vito a bit late, but I resolved the conflicts.

**Comment by ghostsquad** [comment]
I have no idea why these tests are failing. :(

**Comment by ghostsquad** [comment]
@vito I'm going to close this PR, and make a couple more that are smaller in scope

**Comment by ghostsquad** [comment]
Here's one of a few PRs that are smaller in scope to get some of these features through:
https://github.com/concourse/docker-image-resource/pull/218

**Pull request closed without merging**